### PR TITLE
additional test of SendQueue

### PR DESCRIPTION
**Pull request by patriknw**
In my previous attempts to find the reason for the mystery in https://github.com/akka/akka/issues/21624 I wrote this test. I have been running it in a repeat job without luck. It's a good addition anyway.

### CODE REVIEW

File: akka-remote/src/test/scala/akka/remote/artery/SendQueueSpec.scala [hunk 1]
```diff
@@ -147,6 +148,67 @@ class SendQueueSpec extends AkkaSpec("akka.actor.serialize-messages = off") with
       downstream.cancel()
     }
 
+    "deliver first message" in {
+
+      def test(f: (ManyToOneConcurrentArrayQueue[String], SendQueue.QueueValue[String], TestSubscriber.Probe[String]) ⇒ Unit): Unit = {
+
+        (1 to 100).foreach { n ⇒
+          val queue = new ManyToOneConcurrentArrayQueue[String](16)
```
Comment (johanandren) [inline akka-remote/src/test/scala/akka/remote/artery/SendQueueSpec.scala]: But this queue isn't really used anywhere except for passing it to `f`?

Comment (patriknw) [reply akka-remote/src/test/scala/akka/remote/artery/SendQueueSpec.scala]: The test is exercising the different combinations that the queue and the async callback can be used. One way is to add to the queue before injecting the queue into the stage.

Comment (johanandren) [reply akka-remote/src/test/scala/akka/remote/artery/SendQueueSpec.scala]: Aah, now I saw the call where it is injected, must have missed that last time. Thanks for clarifying.

**Review by johanandren — APPROVED** [review]

**Comment by patriknw** [comment]
PLS BUILD

**Pull request merged**